test/e2e/helm_test.go

```diff
@@ -0,0 +1,136 @@
+		Expect(bootstrapClusterProxy).NotTo(BeNil(), "Invalid argument. bootstrapClusterProxy can't be nil when calling %s spec", specName)
+		Expect(os.MkdirAll(artifactFolder, 0o755)).To(Succeed(), "Invalid argument. artifactFolder can't be created for %s spec", specName)
+		Expect(e2eConfig.Variables).To(HaveKey(capi_e2e.KubernetesVersion))
+
+		// CLUSTER_NAME and CLUSTER_NAMESPACE allows for testing existing clusters
+		// if CLUSTER_NAMESPACE is set don't generate a new prefix otherwise
+		// the correct namespace won't be found and a new cluster will be created
+		clusterNameSpace := os.Getenv("CLUSTER_NAMESPACE")
+		if clusterNameSpace == "" {
+			clusterNamePrefix = fmt.Sprintf("caaph-e2e-%s", util.RandomString(6))
+		} else {
+			clusterNamePrefix = clusterNameSpace
+		}
+
+		// Setup a Namespace where to host objects for this spec and create a watcher for the namespace events.
+		var err error
+		namespace, _, err = setupSpecNamespace(ctx, clusterNamePrefix, bootstrapClusterProxy, artifactFolder)
+		// cancelWatches is the blank return arg above.
+		Expect(err).NotTo(HaveOccurred())
+
+		result = new(clusterctl.ApplyClusterTemplateAndWaitResult)
+	})
+
+	Context("Creating workload cluster [REQUIRED]", func() {
+		It("With default template and calico Helm chart", func() {
+			clusterName = fmt.Sprintf("%s-%s", specName, util.RandomString(6))
+			clusterctl.ApplyClusterTemplateAndWait(ctx, createApplyClusterTemplateInput(
+				specName,
+				withNamespace(namespace.Name),
+				withClusterName(clusterName),
+				withControlPlaneMachineCount(1),
+				withWorkerMachineCount(1),
+				withControlPlaneWaiters(clusterctl.ControlPlaneWaiters{
+					WaitForControlPlaneInitialized: EnsureControlPlaneInitialized,
+				}),
+			), result)
+
+			// Create new Helm chart
+			By("Creating new HelmChartProxy to install nginx", func() {
+				hcp := &addonsv1alpha1.HelmChartProxy{
+					TypeMeta: metav1.TypeMeta{
+						APIVersion: addonsv1alpha1.GroupVersion.String(),
+						Kind:       "HelmChartProxy",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "nginx-ingress",
+						Namespace: namespace.Name,
+					},
+					Spec: addonsv1alpha1.HelmChartProxySpec{
+						ClusterSelector: metav1.LabelSelector{
+							MatchLabels: map[string]string{
+								"nginxIngress": "enabled",
+							},
+						},
+						ReleaseName:    "nginx-ingress",
+						ChartName:      "nginx-ingress",
+						RepoURL:        "https://helm.nginx.com/stable",
+						ValuesTemplate: nginxValues,
+						Options:        addonsv1alpha1.HelmOptions{},
+					},
+				}
+				HelmInstallSpec(ctx, func() HelmInstallInput {
+					return HelmInstallInput{
+						BootstrapClusterProxy: bootstrapClusterProxy,
+						Namespace:             namespace,
+						ClusterName:           clusterName,
+						HelmChartProxy:        hcp,
+					}
+				})
+			})
+
+			// TODO: try to update Helm values and expect revision
```

Comment:
IMHO we should implement any `// TODO`s or remove them from code and put into a GitHub issue if they represent future work.

Reply:
I was initially planning on putting it in this PR, which is why it was a TODO. But since this PR is getting rather large, I'll leave it out and open a new issue.